### PR TITLE
Guidance to migrate nanoFramework binding to .NET IoT

### PR DESCRIPTION
**Pull request by jantielens**
## Description

An article to describe the steps to migrate a nanoFramework binding to .NET IoT, it provides guidance in the following areas:
- Fixing the .csproj & .sln files.
- Moving from SpanByte to Span<Byte>.
- Common nanoFramework patterns to more general .NET patters (e.g. infinite while loop).

## Motivation and Context

I created the MPU6886 binding for the nanoFramework and documented the steps I took to make the binding also available in the .NET IoT repo. It could be interesting for developers who want to make their code/binding available to more .NET fans.

## How Has This Been Tested?

NA

## Screenshots
NA

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [x] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: migrate-binding-to-dotnetiot.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+To verify your development environment has the correct .NET SDK's, and all dependencies are available locally it's recommneded to try to build the code. You can do that by running `Build.cmd` (on Windows) or `build.sh` (on Linux) which can be found in the root of the code base.
+
+### 3. Create a new project for your binding and copy code
+
+In the `/src/devices` folder, create a new subfolder for your binding (e.g. by using the name of the sensor).
+
+In this folder , create a new empty .NET project. This can be done with the Visual Studio UI, or by using the command line:
+
+```shell
+dotnet new classlib
```
Comment (Ellerbach) [inline migrate-binding-to-dotnetiot.md:34]: does this create a sln file as well?

**Review by Ellerbach — COMMENTED** [review]
I love it! Thanks for the extra km! Very little remarks.

**Review by josesimoes — APPROVED** [review]
Nice article! Thank your very much for the detailed steps.
I'm sure it will be helpful and will encourage others to port code to .NET IoT!

File: migrate-binding-to-dotnetiot.md (hunk 1 above)
Comment (jantielens) [reply migrate-binding-to-dotnetiot.md:34]: Good catch, Visual Studio will create the .sln (by default), but `dotnet new` doesn't. I'll adjust the next step/comment to fix.

File: README.md [hunk 2]
```diff
@@ -130,6 +130,10 @@ We are using the following structure for the bindings:
 
 The Code Converter allows to facilitate migration of .NET Core/.NET 5.0 code into .NET nanoFramework. More information and how to [customize and run it here](./src/nanoFramework.IoT.Device.CodeConverter/README.md).
 
+## Porting a nanoFramework binding to .NET IoT
```
Comment (Ellerbach) [inline README.md:133]:
```suggestion
## Porting a .NET nanoFramework binding to .NET IoT

```

**Review by Ellerbach — APPROVED** [review]
Great! Thanks again!

**Pull request merged**